src/OrbitDB.js

```diff
@@ -30,17 +31,26 @@ let databaseTypes = {
 class OrbitDB {
   constructor(ipfs, directory, options = {}) {
     this._ipfs = ipfs
-    this.id = options.peerId || (this._ipfs._peerInfo ? this._ipfs._peerInfo.id._idB58String : 'default')
+    this.id = options.peerId
+    this.identity = options.identity
     this._pubsub = options && options.broker
       ? new options.broker(this._ipfs)
       : new Pubsub(this._ipfs, this.id)
-    this.stores = {}
     this.directory = directory || './orbitdb'
-    this.keystore = options.keystore || Keystore.create(path.join(this.directory, this.id, '/keystore'))
-    this.key = this.keystore.getKey(this.id) || this.keystore.createKey(this.id)
+    this.stores = {}
     this._directConnections = {}
   }
 
+  static async init (ipfs, directory, options = {}) {
+    const { id } = await ipfs.id()
+    directory = directory || './orbitdb'
```

Comment:
We could move `directory` to options here, right? If there's no directory provided, we have a default for it and as such, it could be consider *optional*. What do you think? (this would give us the opportunity to eventually move it to optional arg in the constructor, too, in subsequent refactoring)

Reply:
Yea agree we should it to options